keras_nlp/backend/ops.py

```diff
@@ -42,7 +42,8 @@ def take_along_axis(x, indices, axis=None):
             if static_shape[i] != 1:
                 squeezable = False
         if squeezable:
-            indices = tf.squeeze(indices, leftover_axes)
+            if leftover_axes:
+                indices = tf.squeeze(indices, leftover_axes)
             return tf.gather(x, indices, batch_dims=axis)
```

Comment:
wondering - why is the `gather` call in the `if squeezable` branch?

Reply:
This line and its scope is not modified by the PR 🤔 

Might want a separate PR to clean up if it's an issue.

Reply:
Hopefully this is a short lived hack. We would like a `take_along_axis` in keras-core that just works with dynamic shape.

For now the hack is, if indices is all squeezable after `axis`, we can use `gather` directly. Otherwise we would need to write a complex `gather_nd` thing, which would effectively be rewriting `tfnp.take_along_axis`. If we do that, we should just land in keras-core.

Thankfully, this is the last of these workaround we need in `keras-nlp` for the `ops` layer, which is pretty cool!

Reply:
What's being fixed here is a bug in the workaround. I had thought `tf.squeeze(x, axis=())` would be a no-op, but instead it means to squeeze all squeezable axes.

This would break some squeeze calls in the `samplers`, and as a result some of those tests.

My hope is to just spend time patching keras-core itself, rather than the workaround here too much.